### DISCUSSION TITLE
Is the shell supposed to block waiting on response? Should it?

### DISCUSSION THREAD

**Discussion by alexwhittemore** [thonny/thonny#3330] [General]
Hey yall, I'm using Thonny right now for a micropython project that takes input and it seems a lot like every blank line sent from Thonny shell to the device incurs a ~1s delay before a response from the device gets through. See the test cases I wrote up thinking it was a MicroPython problem over here: https://github.com/orgs/micropython/discussions/15913#discussioncomment-10766287. I can reproduce this in Thonny but not any other shell/serial terminal.

What's going on in Thonny that causes this behavior?

**Comment by aivarannamaa** [comment]
Sorry for the late reply! 

I can't reproduce this, but please try again once Thonny 5.0 comes out.
